federationapi/routing/send.go

```diff
@@ -234,17 +235,10 @@ func (t *txnReq) processTransaction(ctx context.Context) (*gomatrixserverlib.Res
 // we should stop processing the transaction, and returns false if it
 // is just some less serious error about a specific event.
 func isProcessingErrorFatal(err error) bool {
-	switch err.(type) {
-	case roomNotFoundError:
-	case *gomatrixserverlib.NotAllowed:
-	case missingPrevEventsError:
-	default:
-		switch err {
-		case context.Canceled:
-		case context.DeadlineExceeded:
-		default:
-			return true
-		}
+	switch err {
+	case sql.ErrConnDone:
+	case sql.ErrTxDone:
```

Comment:
You need to add commentary on why we return 500 in these cases, and why we default to non-fatal errors.

Reply:
Commentary is already at:

https://github.com/matrix-org/dendrite/blob/2bd0449c5b2b5cbefffbe145f6c4c183e4ff0552/federationapi/routing/send.go#L186-L200